test/test_spatialfeature.py

```diff
@@ -193,3 +207,73 @@ def test_feature_contains_positions():
     assert all([a == b for a, b in zip(feature4.contains_positions(positions2),
                                        [False, False, True,
                                         False, True, True])])
+
+
+def test_find_overlapping_cells():
+    t1 = p1.get_overlapping_features(allCells)
+    t2 = p2.get_overlapping_features(allCells)
+    t3 = p3.get_overlapping_features(allCells)
+    t4 = p4.get_overlapping_features(allCells)
+    t5 = p5.get_overlapping_features(allCells)
+
+    assert ((p1 in t1) and (p3 in t1)
+            and (p2 not in t1) and (p5 not in t1) and (p5 not in t1))
+    assert len(t2) == 0
+    assert ((p3 in t3) and (p1 in t3) and (p4 in t3)
+            and (p2 not in t3) and (p5 not in t3))
+    assert ((p4 in t4) and (p3 in t4) and
+            (p1 not in t4) and (p2 not in t4) and (p5 not in t4))
+    assert ((p5 in t5) and (p1 not in t5)
+            and (p2 not in t5) and (p3 not in t5) and (p4 not in t5))
+
+
+def test_remove_overlapping_cells():
+
+    def construct_graph(cells):
+        G = nx.Graph()
+        spatialIndex = rtree.index.Index()
```

Comment:
This is nearly there. I would prefer that the construct_graph function be moved to spatialfeature.py too so that the code doesn't have to be repeated in the tests. This should be possible if the cells are preloaded before calling this function and if _construct_graph in segment.py is modified to take a list of cells and the fov boxes. This will then allow you to expand the tests to make sure that the test cells are allocated to the correct fov.

Reply:
I've addressed this now